Plutarch/Verify.hs

```diff
@@ -0,0 +1,107 @@
+{-# OPTIONS_GHC -Wredundant-constraints #-}
+
+module Plutarch.Verify (
+  PTryFrom (ptryFrom),
+  PTryFromRecur (ptryFromRecur),
+) where
+
+import Plutarch.Builtin (
+  PAsData,
+  PBuiltinList,
+  PBuiltinMap,
+  PBuiltinPair,
+  PData,
+  PIsData (pfromData),
+  pdata,
+  pforgetData,
+  pfstBuiltin,
+  ppairDataBuiltin,
+  psndBuiltin,
+ )
+import Plutarch.ByteString (PByteString)
+import Plutarch.Integer (PInteger)
+import Plutarch.Internal.Other (
+  POpaque,
+  PType,
+  Term,
+  phoistAcyclic,
+  plam,
+  (#),
+  (#$),
+  type (:-->),
+ )
+import Plutarch.List (pmap)
+
+import Plutarch.Lift (PLift)
+import Plutarch.Unsafe (punsafeBuiltin, punsafeCoerce)
+import qualified PlutusCore as PLC
+
+{- |
+    Each POpaque can be of any representation as it represents
+    the universe of Types. With PTryFrom we establish trust
+    in the Data by verifying the requested type matches its
+    representation.
+-}
+class PTryFrom (a :: PType) where
+  ptryFrom :: Term s (PData :--> a)
+
+instance PTryFrom PInteger where
+  ptryFrom = punsafeBuiltin PLC.UnIData
```

Comment:
you should be able to use `pasInt`, `pasList`, and such from `Plutarch.Builtin` in these places

Reply:
Did that, thanks. :)